ranger/ext/img_display.py

```diff
@@ -17,23 +17,43 @@
 import fcntl
 import imghdr
 import os
+import shlex
 import struct
 import sys
 import warnings
 import json
+import mmap
+import shutil
 import threading
-from subprocess import Popen, PIPE
-from collections import defaultdict
+from subprocess import Popen, PIPE, check_call, check_output, CalledProcessError
+from collections import defaultdict, namedtuple
 
 import termios
 from contextlib import contextmanager
 import codecs
-from tempfile import NamedTemporaryFile
+from tempfile import TemporaryFile, NamedTemporaryFile
```

Comment:
I think `TemporaryFile` is more appropriate for this use case, but we could set `dir=…` to address your concerns.

Reply:
What makes it more appropriate? It still gets a name on most file systems, no? Even if it is quickly removed on the happy path it's still nicer to use a sensible name when possible, just in case.

Reply:
A `TemporaryFile` may be invisible from the user's POV. In spite of the name, `NamedTemporaryFile` is not a facility for naming temporary files.

Reply:
I think what @toonn whats to have achieved can be done with TemporaryFile by just prefixing the `prefix` argument with `ranger-`. Then every temporary file that is actually visible in the filesystem tree would be (as far as I understand Python's implementation) be named `ranger-…`, so should anyone find it in the file system it would be named accordingly, if it's invisible to the system it doesn't matter.

If I see this correctly, this would not break your implementation, @remi6397 and @toonn would be happy as well? :)

Reply:
That's… basically what just I said. It's already been done, though with file path rather than "ranger" as the prefix. https://github.com/ranger/ranger/pull/2466/files#diff-4ca0bb8f02a8372446af8e614f9691b6272802bc59968267b66b1e90201b92b7R469

Reply:
@cherti, how about like this? I'm not so sure about the inclusion of the path in the filename anymore, as we could exceed `PATH_MAX`…?

Reply:
ah, sorry, nevermind, somehow I apparently managed to get an outdated version of the diff. nothing said!